### PR TITLE
[FLINK-6367] support custom header settings of allow origin

### PR DESCRIPTION
**Pull request by shijinkui**
`jobmanager.web.access-control-allow-origin`: Enable custom access control parameter for allow origin header, default is `*`.

- [X] General
  - The pull request references the related JIRA issue ("[FLINK-6367] support custom header settings of allow origin")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [X] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [X] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java [hunk 1]
```diff
@@ -701,6 +701,9 @@
 	@Deprecated
 	public static final String JOB_MANAGER_WEB_BACK_PRESSURE_DELAY = "jobmanager.web.backpressure.delay-between-samples";
 
+	/** Web response header of Access-Control-Allow-Origin */
+	public static final String JOB_MANAGER_WEB_ACCESS_CONTROL_ALLOW_ORIGIN = "jobmanager.web.access-control-allow-origin";
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java]: please refactor this into a `ConfigOption`.

Comment (shijinkui) [inline flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java]: Done.

File: flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorConfig.java [hunk 2]
```diff
@@ -78,4 +78,10 @@ public boolean isProgramSubmitEnabled() {
 			ConfigConstants.JOB_MANAGER_WEB_SUBMIT_ENABLED_KEY,
 			ConfigConstants.DEFAULT_JOB_MANAGER_WEB_SUBMIT_ENABLED);
 	}
+
+	public String getAllowOrigin() {
+		return config.getString(
+			ConfigConstants.JOB_MANAGER_WEB_ACCESS_CONTROL_ALLOW_ORIGIN.key(),
```
Comment (zentol) [inline flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorConfig.java]: You can simplify this by using `Configuration#getString(ConfigOption<String> option)` instead.

File: flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java [hunk 3]
```diff
@@ -1333,6 +1333,11 @@
 		key("jobmanager.web.address")
 			.noDefaultValue();
 
+	/** Web response header of Access-Control-Allow-Origin's key and default value. */
+	public static final ConfigOption<String> JOB_MANAGER_WEB_ACCESS_CONTROL_ALLOW_ORIGIN =
+		key("jobmanager.web.access-control-allow-origin").defaultValue("*");
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java]: please move `.defaultValue("*")` into a new line.

File: flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java [hunk 4]
```diff
@@ -1333,6 +1333,11 @@
 		key("jobmanager.web.address")
 			.noDefaultValue();
 
+	/** Web response header of Access-Control-Allow-Origin's key and default value. */
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java]: Please replace this with "The config parameter defining the Access-Control-Allow-Origin header for all responses from the web-frontend".

File: flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java [hunk 5]
```diff
@@ -1333,6 +1333,11 @@
...
+	public static final ConfigOption<String> JOB_MANAGER_WEB_ACCESS_CONTROL_ALLOW_ORIGIN =
+		key("jobmanager.web.access-control-allow-origin").defaultValue("*");
+
+
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java]: Remove this empty line.

**Comment by shijinkui** [comment]
@zentol  fix that

**Comment by zentol** [comment]
Looks good, +1.

**Comment by zentol** [comment]
merging.

**Comment by StephanEwen** [comment]
@zentol Has this been merged? If yes, @shijinkui can you close the pull request manually?

**Comment by shijinkui** [comment]
hi, @zentol @StephanEwen It didn't merge in. @zentol Please check again.

**Comment by zentol** [comment]
Patience please...

**Pull request closed without merging**